### DISCUSSION TITLE
Is 2decomp-fft suitable for large scale 2D FFT?

### DISCUSSION THREAD

**Discussion by hhllhhyyds** [xcompact3d/2decomp-fft#349] [Q&A] [answered]
It seems that this framework is designed to work with large scale 3D FFT, but I want to know if this framework is suitable for large-scale 2D FFT. thanks

**Comment by rfj82982** [comment]
We add the possibility in the current dev branch to skip the FFT in one direction ([see this example](https://github.com/2decomp-fft/2decomp-fft/blob/dev/examples/fft_physical_x/fft_c2c_x_skip.f90)). However input data are always 3D. Just use as 3rd direction 1.

**Reply by hhllhhyyds** [reply]
I see, thanks for your kind interpretation.

**Comment by mathrack** [comment, marked as answer]
Hello,

2decomp-fft can perform large-scale 2D FFT if :

- `nz=1` and the parallel decomposition is `p_row=ncpu` and `p_col=1`
- `nx=1` and the parallel decomposition is `p_row=1` and `p_col=ncpu`

The library is designed for 3D fields. When operating on 2D fields, 2decomp will perform a few useless operations. As pointed out by @rfj82982, some of the useless operations can be skipped.

EDIT : when skipping a direction,  one can use the direction skipped to perform several 2D FFTs simultaneously

Cheers

**Reply by hhllhhyyds** [reply]
I see, thanks for your kind interpretation.
